### PR TITLE
Add Missing Type Annotations

### LINKED ISSUE #15: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_lc709203f.py:58
- [ ] adafruit_lc709203f.py:70
- [ ] adafruit_lc709203f.py:115
- [ ] adafruit_lc709203f.py:143
- [ ] adafruit_lc709203f.py:160
- [ ] adafruit_lc709203f.py:171
- [ ] adafruit_lc709203f.py:182
- [ ] adafruit_lc709203f.py:193
- [ ] adafruit_lc709203f.py:203
- [ ] adafruit_lc709203f.py:210
- [ ] adafruit_lc709203f.py:224
- [ ] adafruit_lc709203f.py:238

### PR DESCRIPTION
**Pull request by tcfranks**
resolves #15 
submitted for review / comment

### CODE REVIEW

File: adafruit_lc709203f.py [hunk 1]
```diff
@@ -55,7 +63,7 @@ class CV:
     """struct helper"""
 
     @classmethod
-    def add_values(cls, value_tuples):
+    def add_values(cls, value_tuples: Tuple[str, int, str, Union[int, None]]) -> None:
```
Comment (tekktrik) [inline adafruit_lc709203f.py]: * This should be wrapped in `Iterable` from `typing`
* The fourth argument should be `Optional[float]`

File: adafruit_lc709203f.py [hunk 2]
```diff
@@ -113,102 +121,102 @@ class LC709203F:
...
 
     @property
-    def ic_version(self):
+    def ic_version(self) -> int:
         """Returns read-only chip version"""
         return self._read_word(LC709203F_CMD_ICVERSION)
 
     @property
-    def power_mode(self):
+    def power_mode(self) -> Literal[0, 1]:
```
Comment (tekktrik) [inline adafruit_lc709203f.py]: This getter and setter should be `Literal[1, 2]` I believe

File: adafruit_lc709203f.py [hunk 3]
```diff
@@ -113,102 +121,102 @@ class LC709203F:
...
     @power_mode.setter
-    def power_mode(self, mode):
+    def power_mode(self, mode: Literal[0, 1]) -> None:
         if not PowerMode.is_valid(mode):
             raise AttributeError("power_mode must be a PowerMode")
         self._write_word(LC709203F_CMD_POWERMODE, mode)
 
     @property
-    def battery_profile(self):
+    def battery_profile(self) -> Literal[0, 1]:
```
Comment (tekktrik) [inline adafruit_lc709203f.py:177]: No change needed here, but here's a good use case where `Literal[0, 1]` makes sense!

File: adafruit_lc709203f.py [hunk 4]
```diff
@@ -113,102 +121,102 @@ class LC709203F:
...
 
     @thermistor_bconstant.setter
-    def thermistor_bconstant(self, bconstant):
+    def thermistor_bconstant(self, bconstant: int) -> None:
         """Sets the thermistor B-constant"""
         self._write_word(LC709203F_CMD_THERMISTORB, bconstant)
 
     @property
-    def thermistor_enable(self):
+    def thermistor_enable(self) -> int:
```
Comment (tekktrik) [inline adafruit_lc709203f.py]: This returns `bool`, see below as well

File: adafruit_lc709203f.py [hunk 5]
```diff
@@ -113,102 +121,102 @@ class LC709203F:
...
         """Returns the current temperature source"""
         return self._read_word(LC709203F_CMD_STATUSBIT)
 
     @thermistor_enable.setter
-    def thermistor_enable(self, status):
+    def thermistor_enable(self, status: Union[Literal[0, 1], bool]) -> None:
```
Comment (tekktrik) [inline adafruit_lc709203f.py]: `status` can be bool, which is the union of `True` and `False`.  A better guard statement, therefore, that can be used here if you want to change it is:

```python
if not isinstance(status, bool):
```

**Review by tekktrik — CHANGES_REQUESTED** [review]
Thanks!  Feedback below:

File: adafruit_lc709203f.py (hunk 5 above)
Comment (tcfranks) [reply adafruit_lc709203f.py]: but not the annotation in the def?

**Comment by tekktrik** [comment]
Feel free to request a review on this one whenever you're ready!

**Comment by tcfranks** [comment]
resubmitted for review with changes per review requests

File: adafruit_lc709203f.py (hunk 5 above)
Comment (tekktrik) [reply adafruit_lc709203f.py]: In the documentation, only the getter's docstring is used.  It's true that it's not super clear what the property does, so we could change it to something like "The enable status of the thermistor as a temperature source".  Does that seem clear?

**Comment by tekktrik** [comment]
One follow up comment above

File: adafruit_lc709203f.py (hunk 5 above)
Comment (tcfranks) [reply adafruit_lc709203f.py]: no, i'm good with the docstring......and on the previous update I changed BOTH the guard test and the passed property to bool, so this is good to review at this point I think.  
Unless you REALLY think the docstring needs an update?

**Review by tekktrik — APPROVED** [review]
Thanks!

**Pull request merged**